application/Common/Shared/Utility/NativeMethods.cs

```diff
@@ -22,18 +32,23 @@ public static class NativeMethods
         /// </summary>
         public static void DoWin32MessageLoop()
         {
-            int status;
-            while ((status = GetMessage(out var msg, IntPtr.Zero, 0, 0)) != 0)
+            // We set to false as we expect a call to this method to be purposeful.
+            shouldCancel = false;
+            while (!shouldCancel)
```

Comment:
The new message Loop makes the CPU usage jump um from almost 0% on the current dev branch to the core being almost fully occupied due to busy waiting
>Unlike GetMessage, the PeekMessage function does not wait for a message to be posted before returning.

As i understand, this loop will not be used on production, so it may be fine for internal testing. I'd still prefer running the blocking loop in other thread if possible.

According to the CPU profiler, the program spends about 97% of its CPU time here (was 4-6% with the old pump using GetMessage).

I also fail to see why we changed from `GetMessage` (which is blocking), to `PeekMessage` in a busy-waiting loop, which essentially makes the `DoWin32MessageLoop` function blocking still.

Reply:
I will try to implement this using get message. However the problem is that GetMessage blocks the thread until it receives a message, so there is no actual way to end the loop. So what I will try to do is to register a custom application message and post it, so I can manually close the loop.